src/library_webgl.js

```diff
@@ -1520,6 +1520,11 @@ for (/**@suppress{duplicate}*/var i = 0; i <= {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
   },
 
   glCompressedTexImage2D: (target, level, internalFormat, width, height, border, imageSize, data) => {
+    // `data` may be null here, which means "allocate space but don't upload" in
```

Comment:
My understanding of the conversation on the issue was that if `data` is null, that's literally just telling GL to read from the null pointer, it has no actual meaning. (glTexImage2D special-cases null, but glCompressedTexImage2D does not.) So seems to me like we should have a debug assert that `data` isn't null.

Reply:
IIUC, `data` is allowed to be null here (i.e. in the GLES C API), and the (in GLES) that means "just allocate some space".  It certainly doesn't mean "read from address zero". since that would instantly crash, right?

I think its just an oversight in the GLES man page that it doesn't document the null special case.  See https://github.com/emscripten-core/emscripten/issues/19300#issuecomment-2305744279: "If the data argument of CompressedTexImage1D, CompressedTexImage2D, or CompressedTexImage3D is NULL, and the pixel unpack buffer object is zero, a texture image with unspecified image contents is created, just as when a NULL pointer is passed to TexImage1D, TexImage2D, or TexImage3D."

Reply:
In GLES3, data is allowed to be null. In GLES2 the specification missed clarifying whether data is allowed to be null or not. https://github.com/KhronosGroup/WebGL/issues/3686

Reply:
Ahh, I didn't read far enough in the GLES3 spec because it was buried way at the bottom under the table!

Reply:
a little clarification:
```suggestion
    // `data` may be null here, which means "allocate uninitialized space but don't upload" in
```